### DISCUSSION TITLE
Prisma Data Proxy connection limit and pool timeout in serverless environment (AWS + Serverless Framework + Prisma ORM)

### DISCUSSION THREAD

**Discussion by mikeSyn** [prisma/prisma#19364] [Q&A] [answered]
Hi, 
I'm trying to use `connection_limit` and `pool_timeout` Prisma Data proxy with specific values, but it seems to ignore them.
I'm using PostgreSQL in AWS RDS.

My current connection test is:

`prisma://aws-eu-central-1.prisma-data.com/?connection_limit=5&pool_timeout=120&api_key=<API_KEY>`

When it occurs an timeout, I get this error:

`Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 3)`

As you can see, the limits are ignored (10 instead of 120 and 3 instead of 5).

We really did a lot of tests. Can anyone help us figuring why?

**Comment by ludralph** [comment, marked as answer]
Hi @mikeSyn 👋 

You would need to set the `pool_timeout` and the `connection_limit` in the database connection string during environment creation. 
for example: "DATABASE_URL=postgresql://aaa:bbb@ccc.us-east-1.rds.amazonaws.com:5432/ddd?connection_limit=10&pool_timeout=100&sslmode=prefer"

Prisma would then use that to generate a connection string to use with the Data Proxy.

**Comment by mikeSyn** [comment]
Hi,
now it seems to work!
Thank you very much!

Mike


Il giorno lun 22 mag 2023 alle ore 17:13 Raphael Etim <
***@***.***> ha scritto:

> Hi @mikeSyn <https://github.com/mikeSyn> 👋
>
> You would need to set the pool_timeout and the connection_limit in the
> database connection string during environment creation.
> for example: "DATABASE_URL=postgresql://aaa:
> ***@***.***
> :5432/ddd?connection_limit=10&pool_timeout=100&sslmode=prefer"
>
> Prisma would then use that to generate a connection string to use with the
> Data Proxy.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/prisma/prisma/discussions/19364#discussioncomment-5968928>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALOMUDXHJ4WYITEHWN7YUTDXHN7BVANCNFSM6AAAAAAYJHTAQI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
--
*michele forchione*
*sr. software engineer*

-- 
 <https://synesthesia.it/>

digital experience company
C.so Dante 118, 
10126 Torino, Italy - tel +39.011.0437401


 
<https://linkedin.com/company/synesthesia>
 
<https://facebook.com/synesthesiait>
 <https://instagram.com/synesthesiait>
 <https://twitter.com/synesthesiait>
 <https://youtube.com/synesthesiait>
Torino ◆ Milano ◆ Modena ◆ synesthesia.it <http://synesthesia.it>




 
<https://synesthesia.it/mail-sign/banner/6f385b0f8196189c81b92552d266b191/syn.gif>
 <https://synesthesia.it/>
